### PR TITLE
allow separate files to be run with mocha

### PR DESCRIPTION
**Pull request by VladVega**
Allow separate files to be run with mocha by adding `before` to every file. Outside of common, require those tests like this: https://github.com/juttle/juttle-postgres-adapter/pull/17

Also:
1. databases tables are now cleaned up after unit tests run.
2. Added functionality to create some or one table.
3. Added check_success function that checks for errors.
4. Fixed a couple of indentation issues

I'm assuming that a +1 here means a +1 on the postgres one as well.

Issue: https://github.com/juttle/juttle-sql-adapter-common/issues/17 and https://github.com/juttle/juttle-sql-adapter-common/issues/27

### CODE REVIEW

**Comment by VladVega** [comment]
@demmer @davidvgalbraith

**Comment by VladVega** [comment]
@demmer I think this is just as we discussed - if you want to take a quick look.

File: test/filters.spec.js [hunk 1]
```diff
@@ -3,6 +3,10 @@ var TestUtils = require("./utils");
 var check_juttle = TestUtils.check_sql_juttle;
 
 describe('test filters', function () {
+    before(function() {
+        TestUtils.init();
+        return TestUtils.loadTables();
```
Comment (demmer) [inline test/filters.spec.js]: Shouldn't you clear the tables in the `after`?

Comment (VladVega) [inline test/filters.spec.js]: I could...but that would be so much slower: to write and delete the tables before every file.

Comment (VladVega) [inline test/filters.spec.js]: Shoot, ok. I didn't realize you could do this globally once for all. Will do.

Comment (VladVega) [inline test/filters.spec.js]: done.

**Comment by VladVega** [comment]
@davidvgalbraith @demmer take a look?

**Comment by davidvgalbraith** [comment]
+1

**Pull request merged**